### ISSUE TITLE
HW3 HEVC / H-265 problem

### ISSUE THREAD

**Issue by BobStrogg** [BobStrogg/teslacam-browser#15]
Newer vehicles with HW3 appear to encode video with the HEVC / H-265 codec.  Unfortunately Chrome (which Electron apps use for the UI) isn't able to handle it, so you'll just see blanks where the videos should be (and clicking "Play" will silently report an error to the console).

**Comment by mwkrieger** [comment]
I don't think this is an HW3 issue. I don't have HW3 on my Model 3 and I'm having this same issue. 

I think it's a software issue because before I upgraded to 2019.16.2 everything worked fine.

**Comment by BobStrogg** [comment]
Hmm, maybe they’re changing everything to HEVC?  Next app update will have option to use default browser rather than embedded Chrome, which may be an adequate workaround for now (Edge and Safari support HEVC without issue).

**Comment by WarbirdXBL** [comment]
Since 2019.16.2, No video playback. 2017 August Model S with HW 2.5.

**Comment by aseniah** [comment]
For now, is there a way we can run the app as a web server and navigate to it from a compatible browser (Safari/Edge)?

**Comment by BobStrogg** [comment]
Next update allows this; hoping to release it today 👍🏻

**Comment by aseniah** [comment]
I'm running the 0.0.9 Mac app now, and it seems to play just fine within the app (I don't need to launch a browser). Not sure if this is using Safari natively, but it seems to be resolved.

**Comment by thienedits** [comment]
This does not work at all. I am on windows 10. Tried electron app and also tried opening via localhost:8088 in Microsoft Edge. Neither plays anything

**Comment by BobStrogg** [comment]
Could you try enabling the developer tools (It’s an option on the View menu), then select the “Console” tab?  There may be some errors shown that could help me diagnose the problem. Thanks!

**Comment by TKGA** [comment]
![image](https://user-images.githubusercontent.com/52500572/60600417-b7174200-9d75-11e9-97c1-4f428aeec829.png)
Not sure if this image comes through, my first time posting on GitHub. I was trying to use the TeslaCam Browser on Windows 10. I have the same problem thienedits has - doesn't play anywhere and just says "0%" for loading. It does pull up the dates and times on the calendar however. I also checked the firewall and everything is open for the app.  I downloaded the latest windows codec and TeslaCam videos do play in Media Player. Any thoughts?

**Comment by BobStrogg** [comment]
Interesting...  If you click the “Browse” button, it should open the same view in your default browser.  Does that work any better? (might have same issue if you use Chrome by default, but pasting address into Edge or Safari should work...)

**Comment by TKGA** [comment]
Hi - yes same issue. I tried Chrome, Edge, and Internet Explorer. On Internet Explorer the UI shows some HTML tags on the page so doesn’t look clean. In Edge it shows the calendar widget correctly but doesn’t show the date/times below the calendar. In Chrome it looks the same as the screenshot (without the console).

**Comment by TKGA** [comment]
Also the flash drive is a Samsung 128gb. I formatted it as FAT32 using 3rd party software. When I plug it in Windows says it needs to be repaired. I decline and am able to open it up just fine. I wasn’t sure if letting windows “repair” it would mess up the drive for the Tesla in any way.

**Comment by TKGA** [comment]
![image](https://user-images.githubusercontent.com/52500572/60608022-0c5a5000-9d84-11e9-9e21-acc16075ccad.png)
![image](https://user-images.githubusercontent.com/52500572/60608078-39a6fe00-9d84-11e9-9f2e-1506bdc5277c.png)
Same problem. First image is the app using a folder on the desktop. Second image is the Edge browser using the folder on the desktop.

**Comment by mitchcapper** [comment]
See #34 for work arounds and further discussion of these two issues.

**Comment by mitchcapper** [comment]
If you are on HW3 please try installing: https://www.microsoft.com/en-us/p/hevc-video-extensions-from-device-manufacturer/9n4wgh0z6vhq  and see if then it works in chrome (or the native electron app).  Another option is to use firefox /edge to manually use the web service (use launch browser and copy that url to one of those browsers).   Please reply if those items not fix the issue.  If they do fix the issue please just  thumbs up this reply.

**Comment by thienedits** [comment]
Hey, just wanted to give another option. I created a site for viewing the video files. It works in Safari and Microsoft Edge, however Edge is slow to load when you load a lot of files. I installed the new Chromium version of Microsoft Edge and it works and loads much quicker. You can download the chromium version here https://www.microsoftedgeinsider.com/en-us/

My TeslaCam site:
https://viewmytesla.appspot.com/

**Comment by mitchcapper** [comment]
It is interesting microsoft edge would work without the store plugin are you 100% sure it works with HW3 HEVC videos without it?   The good news is if that is a case you can use with teslacam browser as well as you can use any browser with the local server.

**Comment by thienedits** [comment]
Yeah I find it weird the chromium Edge works whereas chrome did not. I do have the HEVC plugin installed so that may be why it works. TeslaCam browser did not work on chromium Edge so it may be some other issue with your app.

**Comment by BobStrogg** [comment]
I'm assuming it's because Microsoft is ok paying the HEVC licensing fee? (I assume same reason previous Edge browser worked, and similarly with Apple & Safari).  Google would probably prefer AV1 or VP9 to be broadly adopted.

**Comment by thienedits** [comment]
Yea I think google is developing VP9 codec because it is royalty free whereas H.265 is not.
